api/routes/generate.js

```diff
@@ -47,149 +42,187 @@ exports.buildCrxConfig = function *(targetUrl) {
         filename: parsedUrl.hostname + '.crx'
     };
 
-    // Prepare request (send fake browser user-agent header)
-    var req = {
-        url: crxConfig.url,
-        headers: {
-            'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.106 Safari/537.36'
-        }
-    };
-    
-    // Page HTML response
-    var res;
-    
+
     // Execute GET request to provided URL
     // May fail for internal URLs, continue anyway
     try {
-        res = yield thunkify(request)(req);
+        var dom = yield exports.getPageDom(crxConfig.url);
     }
-    catch(exc) {
-        // Return fallback config
-        return crxConfig;
+    catch (exc) {
+        console.log("Failed to get page dom")
     }
-    
-    // Load DOM into Cheerio (HTML parser)
-    var dom = cheerio.load(res[0].body);
-    
-    // Extract extension title from the dom's <title> tag
-    crxConfig.title = dom('title').text().trim() || crxConfig.parsedUrl.hostname;
-    
-    // Extract .crx icon from page's shortcut-icon <link> element
-    crxConfig.icon = dom('link[rel="icon"], link[rel="shortcut icon"]').attr('href');
-     
+
+    if (dom) {
+        crxConfig.title = exports.getExtensionTitle(dom, crxConfig.parsedUrl.hostname);
+
+        crxConfig.icon = exports.getExtensionIcon(dom, crxConfig.parsedUrl.host);
```

Comment:
According to the [Node.js docs](https://nodejs.org/docs/latest/api/url.html#url_url_parsing), `host` contains the port number and `hostname` does not. I think that we should use `host` in this case then, in case users will generate several extensions for the same hostname, on different ports.